### PR TITLE
CI: pylint to ruff

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -186,3 +185,10 @@ opencv-python-headless = { url = "https://github.com/commaai/opencv-python-build
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
+
+# https://beta.ruff.rs/docs/configuration/#using-pyprojecttoml
+[tool.ruff]
+select = ["E", "F", "W"]
+ignore = ["W292", "E741"]
+line-length = 2500
```
Comment (adeebshihadeh) [inline pyproject.toml:193]: why was this changed?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

**Comment by adeebshihadeh** [comment]
Why was this merged and the comment marked as resolved?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -34,31 +34,11 @@ repos:
...
-        language: system
-        types: [python]
-        exclude: '^(third_party/)|(cereal/)|(rednose/)|(panda/)|(laika/)|(laika_repo/)|(rednose_repo/)|(tinygrad/)|(tinygrad_repo/)'
-        args:
-        - -j0
-        - -rn
-        - -sn
-        - --rcfile=.pylintrc
+    -   id: ruff
+        exclude: '^(third_party/)|(cereal/)|(body/)|(rednose/)|(rednose_repo/)|(opendbc/)|(panda/)|(tools/)|(selfdrive/modeld/thneed/debug/)|(selfdrive/modeld/test/)|(selfdrive/camerad/test/)|(installer/)|(tinygrad_repo/)'
```
Comment (adeebshihadeh) [inline .pre-commit-config.yaml:41]: why is tools excluded? the only exclusions should be submodules, which are tested in their own repo, and third party code

**Comment by jnewb1** [comment]
> Why was this merged and the comment marked as resolved?

@adeebshihadeh moving to +160 line to another PR because I think it will be cleaner
https://github.com/commaai/openpilot/pull/29279
